### PR TITLE
feat: Update (and streamline) the default runtime to be DOTNET_8

### PR DESCRIPTION
**Pull request by lachriz-aws**
The default runtime was set to `DOTNET_6` while the docs said `PROVIDED_AL2`. This PR streamlines this, and changes the default runtime to be `DOTNET_8`.

### CODE REVIEW

File: src/function.ts [hunk 1]
```diff
@@ -80,7 +79,7 @@ export class DotNetFunction extends lambda.Function {
           : lambda.RuntimeFamily.OTHER;
         runtime = new lambda.Runtime(toolsRuntime, family);
       } else {
-        runtime = lambda.Runtime.DOTNET_6;
+        runtime = lambda.Runtime.DOTNET_8;
```
Comment (vlesierse) [inline src/function.ts:82]: Currently there is a test `integ.function-dotnet8` which tests explicitly set runtimes. With this change I would to suggest to change the test to `DOTNET_6` until we drop support in November.

**Review by vlesierse — CHANGES_REQUESTED** [review]
With .NET 8 as default and the out of support of .NET 6 in November, this fix makes sense. It is a breaking change though, so we should mark it as features or include this in the other PR #259 with a minor version bump.

**Comment by lachriz-aws** [comment]
> With .NET 8 as default and the out of support of .NET 6 in November, this fix makes sense. It is a breaking change though, so we should mark it as features or include this in the other PR #259 with a minor version bump.

Now that #259 is marked as a `fix`, I think it would make sense to follow your proposal of changing this PR (#260) to be a feature - and hence have them tracked as two different changes. I will change that now.

In terms of versioning - the current version is `0.0.4`. Reading you suggestion to bump the minor version -- from a semantic versioning point of view, that would imply we do a `0.1.0` release, wouldn't it? Alternatively, we could also consider discussing if we instead want to do a first stable release (`1.0.0`) instead and have this PR included as part of that release?`

Many options to choose from - let me know what you think.

File: src/function.ts (hunk 1 above)
Comment (lachriz-aws) [reply src/function.ts:82]: Yes, I agree with that. Will do that change as well.

**Comment by vlesierse** [comment]
@lachriz-aws we can leave both PR separate. For versioning, currently this library is marked as experimental by the CDK team. We can take this to them on what is needed to bring it to a stable 1.0 release. Meanwhile following semantic versioning 0.1.0 would be a good next version for this feature.

File: src/function.ts (hunk 1 above)
Comment (lachriz-aws) [reply src/function.ts:82]: Now completed ✅

**Comment by lachriz-aws** [comment]
Thanks for the above feedback @vlesierse. The PR now contains the following changes:

- The default runtime is now .NET 8
- The `lambda-handler-aot` integ test project has been upgraded to .NET 8 (from .NET 7 which is end of life).
- The `lambda-handler-aot` integ test project took taking a dependency on libicu (I opted for

**Review by vlesierse — APPROVED** [review]
LGTM

**Pull request merged**